mm2src/coins/z_coin/z_params/indexeddb.rs

```diff
@@ -143,13 +143,20 @@ impl ZcashParamsWasmImpl {
 
     /// Download and save z_params to storage.
     pub(crate) async fn download_and_save_params(&self) -> MmResult<(Vec<u8>, Vec<u8>), ZcoinStorageError> {
-        let (sapling_spend, sapling_output) = super::download_parameters().await.unwrap();
+        let (sapling_spend, sapling_output) = super::download_parameters()
+            .await
+            .mm_err(|err| ZcoinStorageError::ZcashParamsError(err.to_string()))?;
         let spends = sapling_spend_to_chunks(&sapling_spend);
-        for (i, spend) in spends.into_iter().enumerate() {
-            if i == 0 {
-                self.save_params(i as u8, &spend, &sapling_output).await?
-            } else {
-                self.save_params(i as u8, &spend, &[]).await?
+
+        if sapling_spend.len() <= sapling_output.len() {
+            self.save_params(0, &sapling_spend, &sapling_output).await?
+        } else {
+            for (i, spend) in spends.into_iter().enumerate() {
+                if i == 0 {
+                    self.save_params(i as u8, spend, &sapling_output).await?
+                } else {
+                    self.save_params(i as u8, spend, &[]).await?
+                }
             }
         }
```

Comment:
You can create sapling chunks later in `else` block, and get `first spend chunk` using `split_first()`, so you wont need to do i==0 check every iteration
```rust
        let (sapling_spend, sapling_output) = super::download_parameters()
            .await
            .mm_err(|err| ZcoinStorageError::ZcashParamsError(err.to_string()))?;

        if sapling_spend.len() <= sapling_output.len() {
            // If sapling_spend is not larger than sapling_output, save it directly
            self.save_params(0, &sapling_spend, &sapling_output).await?;
        } else {
            let spends = sapling_spend_to_chunks(&sapling_spend);

            if let Some((first_spend, remaining_spends)) = spends.split_first() {
                self.save_params(0, first_spend, &sapling_output).await?;

                for (i, spend) in remaining_spends.iter().enumerate() {
                    self.save_params((i + 1) as u8, spend, &[]).await?;
                }
            }
        }
```